networkit/test/test_vizbridges.py

```diff
@@ -0,0 +1,148 @@
+                    showIds=False,
+                )
+
+    def testVizCustomSize(self):
+        for dim, directed, weighted in zip(
+            vizbridges.Dimension, [True, False], [True, False]
+        ):
+            with self.subTest(dim=dim, directed=directed, weighted=weighted):
+                G = self.getSmallGraph(weighted, directed)
+                vizbridges.widgetFromGraph(
+                    G,
+                    dimension=dim,
+                    nodeScores=list(range(G.numberOfNodes())),
+                    customSize=50,
+                )
+
+    def testVizEdgeScoresList(self):
+        for dim, directed, weighted in zip(
+            vizbridges.Dimension, [True, False], [True, False]
+        ):
+            with self.subTest(dim=dim, directed=directed, weighted=weighted):
+                G = self.getSmallGraph(weighted, directed)
+                G.indexEdges()
+                vizbridges.widgetFromGraph(
+                    G,
+                    dimension=dim,
+                    edgeScores=list(range(G.numberOfEdges())),
+                )
+
+    def testVizEdgeScoresDict(self):
+        for dim, directed, weighted in zip(
+            vizbridges.Dimension, [True, False], [True, False]
+        ):
+            with self.subTest(dim=dim, directed=directed, weighted=weighted):
+                G = self.getSmallGraph(weighted, directed)
+                G.indexEdges()
+                scores = {}
+                for i, [u, v] in enumerate(G.iterEdges()):
+                    scores[u, v] = i
+                vizbridges.widgetFromGraph(
+                    G,
+                    dimension=dim,
+                    edgeScores=scores,
+                )
+
+    def testVizEdgeScoresDefaultdict(self):
+        for dim, directed, weighted in zip(
+            vizbridges.Dimension, [True, False], [True, False]
+        ):
+            with self.subTest(dim=dim, directed=directed, weighted=weighted):
+                G = self.getSmallGraph(weighted, directed)
+                G.indexEdges()
+                scores = defaultdict(lambda: 0)
+                for u, v in G.iterEdges():
+                    scores[u, v] = 1
+                    break
+                vizbridges.widgetFromGraph(
+                    G,
+                    dimension=dim,
+                    edgeScores=scores,
+                )
+
+    def testVizEdgePalette(self):
+        for dim, directed, weighted in zip(
+            vizbridges.Dimension, [True, False], [True, False]
+        ):
+            with self.subTest(dim=dim, directed=directed, weighted=weighted):
+                G = self.getSmallGraph(weighted, directed)
+                G.indexEdges()
+                vizbridges.widgetFromGraph(
+                    G,
+                    dimension=dim,
+                    edgeScores=list(range(G.numberOfEdges())),
+                    edgePalette=[(i / 10, i / 10, i / 10) for i in range(4)],
+                )
+
```

Comment:
missing/todo: tests for incomplete/invalid inputs like incomplete score vectors and mutually exclusive arguments.